### PR TITLE
Adding Unique ID for Sensors

### PR DESCRIPTION
**Pull request by troykelly**
Sensors need a `unique_id` per [here](https://developers.home-assistant.io/docs/entity_registry_index/).

### CODE REVIEW

**Comment by troykelly** [comment]
Screen Shot 2021-07-14 at 4 45 51 pm

I've got a `PR` in for [wheel](https://github.com/home-assistant/wheels-custom-integrations/pull/460).
`brand` can just be a rename of amber_electric [here](https://github.com/home-assistant/brands/tree/master/custom_integrations).

File: custom_components/amberelectric/sensor.py [hunk 1]
```diff
@@ -40,6 +41,11 @@ def __init__(self, platform_name: str, channel_type: str, data_service: AmberDat
     def name(self) -> Union[str, None]:
         return self._platform_name + " - " + friendly_channel_type(self._channel_type) + " " + " Price"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the device"""
+        return "%s_%s_%s" % (self._platform_name, self._channel_type, "Price")
```
Comment (madpilot) [inline custom_components/amberelectric/sensor.py:47]: `_platform_name` isn't guaranteed to be unique, and isn't immutable. Better to use site id, which will require a bit of a refactor so `AmberPriceSensor` takes it as a parameter.

Also, use slugify (`from homeassistant.util import slugify` when generating the string.

File: custom_components/amberelectric/sensor.py [hunk 2]
```diff
@@ -96,6 +102,11 @@ def __init__(self, platform_name: str, data_service: AmberDataService) -> None:
     def name(self) -> Union[str, None]:
         return self._platform_name + " - Renewables"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the device"""
+        return "%s_%s" % (self._platform_name, "Renewables")
```
Comment (madpilot) [inline custom_components/amberelectric/sensor.py:108]: See above comment

File: custom_components/amberelectric/sensor.py [hunk 3]
```diff
@@ -128,6 +139,11 @@ def __init__(self, platform_name: str, channel_type: str, data_service: AmberDat
     def name(self) -> Union[str, None]:
         return self._platform_name + " - " + friendly_channel_type(self._channel_type) + " " + " Forecast"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the device"""
+        return "%s_%s_%s" % (self._platform_name, self._channel_type, "Forecast")
```
Comment (madpilot) [inline custom_components/amberelectric/sensor.py:145]: See above comment

File: custom_components/amberelectric/sensor.py [hunk 4]
```diff
@@ -188,6 +204,11 @@ def __init__(self, platform_name: str, data_service: AmberDataService) -> None:
     def name(self) -> Union[str, None]:
         return self._platform_name + " - Price Spike"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the device"""
+        return "%s_%s" % (self._platform_name, "PriceSpike")
```
Comment (madpilot) [inline custom_components/amberelectric/sensor.py:210]: See above comment

**Review by madpilot — CHANGES_REQUESTED** [review]
Thanks for the PR - a couple of small changes around the unique_id (See comments below)

**Comment by madpilot** [comment]
I have no idea what the "The repository is missing topics" error means. Any ideas? If not, can you add an ignore: https://github.com/hacs/action#ignorable-checks

**Comment by madpilot** [comment]
I actually just did this: https://github.com/madpilot/hass-amber-electric/commit/e6783e6303134e9be1761c8a2f9c7e573632954d

**Comment by troykelly** [comment]
@madpilot Thank you!
I wasn't sure when I'd get a minute to get back to this.
Great work with the integration.

**Pull request closed without merging**